code/src/components/ShowManualInput.js

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { fetchProduct } from "../reducers/products"
+import styled from 'styled-components'
+
+export const ShowManualInput = () => {
+  const [showInput, setInput] = useState(false)
+  const [code, setCode] = useState("")
+  const dispatch = useDispatch()
+
+  const handleSubmit = e => {
+    e.preventDefault()
+    dispatch(fetchProduct(code))
+    setCode("")
+  }
+
+  return (
+    <>
+      {!showInput && (
+        <Wrapper>
+          <Text>Or enter the product code number manually:</Text>
+          <Button type="button" onClick={() => setInput(true)}><ButtonText>manual</ButtonText>
+          </Button>
+        </Wrapper>
+      )}
+      {showInput && (
+        <form onSubmit={handleSubmit}>
+          <Input
+            type='text'
+            value={code}
+            onChange={e => setCode(e.target.value)}
+            placeholder='Enter barcode number...'
+            required
+          />
```

Comment:
Nice with a form fallback for if the camera scanner approach isn't working well!